docs/usage/general.rst

```diff
@@ -31,6 +31,20 @@ All Borg commands share these options:
 
 .. include:: common-options.rst.inc
 
+Option ``--bypass-lock`` allows you to access the repository while bypassing
```

Comment:
i thought about the option name while reading through the changes.

i rather think it should be called `--read-only-repo` or so, because it does not only suppress lock creation but also all kinds of other write ops to the repo dir.

Reply:
Hmm, yes, I understand where you're coming from. However, two counter points to consider:

- I went with `--bypass-lock` because I thought that describes most accurately what happens: it actually ignores and bypasses the locking mechanism. If you'd use `borg with-lock` to indefinitely keep a lock on the repo, you could still access it with `--bypass-lock`. Using this "bypassing feature" is just a tool to also provide access a read-only repo.
- > i rather think it should be called --read-only-repo or so, because it does not only suppress lock creation but also all kinds of other write ops to the repo dir

    The option (and the changeset it introduces) actually doesn't suppress other writing operations at all. It only suppresses errors if there are permission conflicts. If the repo is not read-only, there is effectively no difference between using `--bypass-lock` or not. (Apart from not using a lock for the operations.)

Reply:
Oh, you meant "suppress write operations" because of the manually raised exception?

Reply:
Do you mind clarifying what you meant?

I deliberately chose `NotImplementedError` in [those respective lines](https://github.com/borgbackup/borg/pull/5048/files#diff-48f2b9140a35ffd45a46ccfbe63931a6R4344-R4350) which check for "not allowed commands". Would you rather omit that check altogether and have the user simply get the inevitable `PermissionError` if they're using `--bypass-lock` with a write command like `borg create`? (Note, it won't be `LockFailed`, but `PermissionError`.)

Reply:
looking at this again now...

Reply:
well, i rather meant "silently fail" than "suppress" (e.g. in save_config).

i am wondering a bit about the nonce and the start transaction stuff - is that used by any of the supported commands? sounds more like for repo-writing ops.

Reply:
About the NotImplementedError exception in case of an unsupported command: we should exit cleanly in such a case. no long traceback as this is expected and we do not need to analyze a bug here.

Reply:
> i am wondering a bit about the nonce and the start transaction stuff - is that used by any of the supported commands?

Yes, `borg check` uses those. There is a call to `save_config` [here](https://github.com/borgbackup/borg/blob/master/src/borg/repository.py#L979) for the repository check. And a call to `commit_nonce_reservation` happens during `rebuild_refcounts` [here](https://github.com/borgbackup/borg/blob/master/src/borg/archive.py#L1427) for the archives check.

Reply:
> About the NotImplementedError exception in case of an unsupported command: we should exit cleanly in such a case. no long traceback as this is expected and we do not need to analyze a bug here.

What does that mean, effectively? :)

Reply:
try: `raise Error("error message for user")`

Reply:
ok, the not working save_config (for read-only case) means that incremental checks won't work, which is expected if we can't remember how far we got.

the rebuild_refcounts stuff is only commit in repair mode.

Reply:
`helpers.Error` calls `format` but `__doc__` doesn't contain any `{}` :thinking:  https://github.com/borgbackup/borg/blob/master/src/borg/helpers/errors.py#L6

Reply:
guess it is because we only used subclasses of this yet (or did not give a msg).

Reply:
Two occurrences:

- https://github.com/borgbackup/borg/blob/master/src/borg/archiver.py#L1707
- https://github.com/borgbackup/borg/blob/master/src/borg/helpers/manifest.py#L107

Reply:
Those two occurrences basically never happen - still shouldn't be like this, no? Concerns ErrorWithTraceback, too:

```
>>> raise ErrorWithTraceback("custom message")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
__main__.ErrorWithTraceback: like Error, but show a traceback also
```

Opened a pull request to fix it. #5100